com.unity.render-pipelines.core/Runtime/XR/XRBuiltinShaderConstants.cs

```diff
@@ -0,0 +1,72 @@
+namespace UnityEngine.Rendering
+{
+    /// <summary>
+    /// Helper static class used by render pipelines to setup stereo constants accessed by builtin shaders.
+    /// </summary>
+    public static class XRBuiltinShaderConstants
+    {
+        // References to builtin shader constants
+        static readonly int k_StereoCameraProjection    = Shader.PropertyToID("unity_StereoCameraProjection");
+        static readonly int k_StereoCameraInvProjection = Shader.PropertyToID("unity_StereoCameraInvProjection");
+        static readonly int k_StereoMatrixV             = Shader.PropertyToID("unity_StereoMatrixV");
+        static readonly int k_StereoMatrixInvV          = Shader.PropertyToID("unity_StereoMatrixInvV");
+        static readonly int k_StereoMatrixP             = Shader.PropertyToID("unity_StereoMatrixP");
+        static readonly int k_StereoMatrixInvP          = Shader.PropertyToID("unity_StereoMatrixInvP");
+        static readonly int k_StereoMatrixVP            = Shader.PropertyToID("unity_StereoMatrixVP");
+        static readonly int k_StereoMatrixInvVP         = Shader.PropertyToID("unity_StereoMatrixInvVP");
+        static readonly int k_StereoWorldSpaceCameraPos = Shader.PropertyToID("unity_StereoWorldSpaceCameraPos");
+
```

Comment:
Since built-in shaders will be gradually removed from URP (ie. Skybox), this built-in shader constants file should remain in URP instead of promoting to Core. Moving to Core would only confuse developers to rely more on built-in constants instead of URP ones.  HDRP also don't use built-in so there's no need to share this file with HDRP.

Reply:
The built-in constants are required on both pipelines for backwards compatibility with older shaders. I don't think we can get rid of them until we get rid of the builti-in pipeline.